src/Microsoft.Framework.ConfigurationModel/Configuration.cs

```diff
@@ -13,7 +13,7 @@ public class Configuration : IConfiguration, IConfigurationSourceRoot
     {
         private readonly IList<IConfigurationSource> _sources = new List<IConfigurationSource>();
 
-        public Configuration(params IConfigurationSource[] sources)
+        public Configuration(string basePath, params IConfigurationSource[] sources)
```

Comment:
I'll allow this to be a null value and throw exception in later stage if a file based configuration source can't resolve an absolute path without the base path.